experimental/FTheoryTools/src/AbstractFTheoryModels/methods.jl

```diff
@@ -0,0 +1,547 @@
+  S = cox_ring(new_ambient_space)
+  _e = eval_poly(coordinate_name, S)
+  images = MPolyRingElem[]
+  for v in gens(S)
+    v == _e && continue
+    if string(v) in [string(k) for k in gens(I)]
+      push!(images, v * _e)
+    else
+      push!(images, v)
+    end
+  end
+  ring_map = hom(base_ring(I), S, images)
+
+  # Construct the new model
+  if m isa GlobalTateModel
+    new_pt = _my_proper_transform(ring_map, tate_polynomial(m), coordinate_name)
+    model = GlobalTateModel(explicit_model_sections(m), defining_section_parametrization(m), new_pt, base_space(m), new_ambient_space)
+  else
+    new_pw = _my_proper_transform(ring_map, weierstrass_polynomial(m), coordinate_name)
+    model = WeierstrassModel(explicit_model_sections(m), defining_section_parametrization(m), new_pw, base_space(m), new_ambient_space)
+  end
+
+  # Copy/overwrite known attributes from old model
+  model_attributes = m.__attrs
+  for (key, value) in model_attributes
+    set_attribute!(model, key, value)
+  end
+  set_attribute!(model, :partially_resolved, true)
+
+  # Return the model
+  return model
+end
+
+
+@doc raw"""
+    tune(m::AbstractFTheoryModel, p::MPolyRingElem; completeness_check::Bool = true)
+
+Tune an F-theory model by replacing the hypersurface equation by a custom (polynomial)
+equation. The latter can be any type of polynomial: a Tate polynomial, a Weierstrass
+polynomial or a general polynomial. We do not conduct checks to tell which type the
+provided polynomial is. Consequently, this tuning will always return a hypersurface model.
+
+Note that there is less functionality for hypersurface models than for Weierstrass or Tate
+models. For instance, `singular_loci` can (currently) not be computed for hypersurface models.
+
+# Examples
+```jldoctest
+julia> B3 = projective_space(NormalToricVariety, 3)
+Normal toric variety
+
+julia> w = torusinvariant_prime_divisors(B3)[1]
+Torus-invariant, prime divisor on a normal toric variety
+
+julia> t = literature_model(arxiv_id = "1109.3454", equation = "3.1", base_space = B3, model_sections = Dict("w" => w), completeness_check = false)
+Construction over concrete base may lead to singularity enhancement. Consider computing singular_loci. However, this may take time!
+
+Global Tate model over a concrete base -- SU(5)xU(1) restricted Tate model based on arXiv paper 1109.3454 Eq. (3.1)
+
+julia> x1, x2, x3, x4, x, y, z = gens(parent(tate_polynomial(t)))
+7-element Vector{MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}}:
+ x1
+ x2
+ x3
+ x4
+ x
+ y
+ z
+
+julia> new_tate_polynomial = x^3 - y^2 - x * y * z * x4^4
+-x4^4*x*y*z + x^3 - y^2
+
+julia> tuned_t = tune(t, new_tate_polynomial)
+Hypersurface model over a concrete base
+
+julia> hypersurface_equation(tuned_t) == new_tate_polynomial
+true
+
+julia> base_space(tuned_t) == base_space(t)
+true
+```
+"""
+function tune(m::AbstractFTheoryModel, p::MPolyRingElem; completeness_check::Bool = true)
```

Comment:
I'm probably missing something here, but this doesn't do any checks that the resulting model with the new hyper surface equation is in any way related to the original model right? Isn't this just creating a completely new model?

Reply:
Correct. The provided polynomial can be very different from the original one. For instance, you could start with a Tate model and tune the equation such that it is no longer a Tate model. Hence, this method will return a hypersurface model.

Of course, if we are more careful, then the tuned model will be similar to the original model. But this seems to be subtle. Right now, I do not even dare to try to tell if the provided polynomial is chosen "carefully".

Reply:
But of course, this can be addressed in the future.